### PR TITLE
obis-observer: add commands

### PR DESCRIPTION
**Pull request by DeadByElpy**

### CODE REVIEW

File: src/obis-observer/commands/uplink/setObserverSingleMode.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/**
+ * Response to set the single or multi mode of the observer device.
+ *
+ * @example create command instance from command body hex dump
+ * ```js
+ * import * as setObserverSingleMode from 'jooby-codec/obis-observer/commands/uplink/setObserverSingleMode.js';
+ *
+ * const bytes =  [0x07];
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/setObserverSingleMode.ts:8]:
```suggestion
 * const bytes = [0x07];
```

Comment (DarkPark) [reply src/obis-observer/commands/uplink/setObserverSingleMode.ts:8]: and below

File: src/obis-observer/commands/uplink/setSerialPort.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+/**
+ * Response to set serial port parameters.
+ *
+ * @example create command instance from command body hex dump
+ * ```js
+ * import * as setSerialPort from 'jooby-codec/obis-observer/commands/uplink/setSerialPort.js';
+ *
+ * const bytes = [0x20];
+ * const parameters =  setSerialPort.fromBytes(bytes);
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/setSerialPort.ts:9]:
```suggestion
 * const parameters = setSerialPort.fromBytes(bytes);
```

File: src/obis-observer/commands/downlink/setupMeter.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+/**
+ * Encode command parameters.
+ *
+ * @param parameters - command payload
+ * @returns full message (header with body)
+ */
+export const toBytes = ( parameters: ISetupMeterParameters ): types.TBytes => {
+    const buffer: ICommandBinaryBuffer = new CommandBinaryBuffer(getCommandSize(parameters));
+    const {requestId, meterId, meterProfileId, address} = parameters;
+
```
Comment (DarkPark) [inline src/obis-observer/commands/downlink/setupMeter.ts:157]:
```suggestion
```

File: src/obis-observer/commands/downlink/setupMeter.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    const isMeterProfileIdExist = meterProfileId || meterProfileId === 0;
+    buffer.setUint8(requestId);
```
Comment (DarkPark) [inline src/obis-observer/commands/downlink/setupMeter.ts:159]:
```suggestion

    buffer.setUint8(requestId);
```

File: src/obis-observer/commands/downlink/setupMeter.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+    buffer.setUint32(meterId);
+
+    if ( address.length !== 0 || isMeterProfileIdExist ) {
+        buffer.setString(address);
+        if (isMeterProfileIdExist) {
```
Comment (DarkPark) [inline src/obis-observer/commands/downlink/setupMeter.ts:164]:
```suggestion

        if ( isMeterProfileIdExist ) {
```

File: src/obis-observer/commands/uplink/setupMeter.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
+/**
+ * Response to setup the meter id, meter profile id, meter address to the meter. A new meter will be created if it doesn't exist.
+ *
+ * @example create command instance from command body hex dump
+ * ```js
+ * import * as setupMeter from 'jooby-codec/obis-observer/commands/uplink/setupMeter.js';
+ *
+ * const bytes = [0x07];
+ * const parameters =  setupMeter.fromBytes(bytes);
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/setupMeter.ts:9]:
```suggestion
 * const parameters = setupMeter.fromBytes(bytes);
```

File: src/obis-observer/commands/uplink/setupMeterProfile.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
+/**
+ * Response to setup the meter profile.
+ *
+ * @example create command instance from command body hex dump
+ * ```js
+ * import * as setupMeterProfile from 'jooby-codec/obis-observer/commands/uplink/setupMeterProfile.js';
+ *
+ * const bytes =  [0x9c];
+ * const parameters =  setupMeterProfile.fromBytes(bytes);
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/setupMeterProfile.ts:9]:
```suggestion
 * const parameters = setupMeterProfile.fromBytes(bytes);
```

File: src/obis-observer/commands/uplink/updateImageVerify.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+    return {requestId, isImageValid};
+};
+
+/**
+ * Encode command parameters.
+ *
+ * @returns full message (header with body)
+ */
+export const toBytes = (parameters: IUpdateImageVerifyParameters): types.TBytes => command.toBytes(id, [parameters.requestId, parameters.isImageValid ? 1 : 0]);
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/updateImageVerify.ts:74]:
```suggestion
export const toBytes = (parameters: IUpdateImageVerifyParameters): types.TBytes => command.toBytes(
    id,
    [parameters.requestId, parameters.isImageValid ? 1 : 0]
);
```

?

File: src/obis-observer/commands/downlink/updateImageWrite.ts [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+ * @packageDocumentation
+ *
+ * @example
+ * ```js
+ * import * as updateImageWrite from 'jooby-codec/obis-observer/commands/downlink/updateImageWrite.js';
+ *
+ * const parameters = {
+ *     requestId: 33,
+ *     offset: 2112,
+ *     data: new Uint8Array([
```
Comment (DarkPark) [inline src/obis-observer/commands/downlink/updateImageWrite.ts:13]: wrong example

File: src/obis-observer/commands/downlink/updateRun.ts [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+ * [Command format documentation](https://github.com/jooby-dev/jooby-docs/blob/main/docs/obis-observer/commands/UpdateImageWrite.md#request)
+ */
+
+import * as command from '../../utils/command.js';
+import * as types from '../../../types.js';
+import CommandBinaryBuffer, {ICommandBinaryBuffer, ICommandParameters} from '../../utils/CommandBinaryBuffer.js';
+
+
+interface IUpdateRunParameters extends ICommandParameters {
+    requestId: types.TUint8
```
[the remark this answers is not in the document]
Comment (DeadByElpy) [reply src/obis-observer/commands/downlink/updateRun.ts:30]: fixed

File: src/obis-observer/commands/downlink/updateImageWrite.ts (hunk 9 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/downlink/updateImageWrite.ts:13]: fixed

File: src/obis-observer/commands/uplink/updateImageVerify.ts (hunk 8 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/uplink/updateImageVerify.ts:74]: fixed

File: src/obis-observer/commands/uplink/setupMeterProfile.ts (hunk 7 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/uplink/setupMeterProfile.ts:9]: fixed in all places

File: src/obis-observer/commands/uplink/setupMeter.ts (hunk 6 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/uplink/setupMeter.ts:9]: fixed in all places

File: src/obis-observer/commands/downlink/setupMeter.ts (hunk 5 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/downlink/setupMeter.ts:164]: fixed in all places

File: src/obis-observer/commands/downlink/setupMeter.ts (hunk 4 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/downlink/setupMeter.ts:159]: fixed

File: src/obis-observer/commands/downlink/setupMeter.ts (hunk 3 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/downlink/setupMeter.ts:157]: fixed

File: src/obis-observer/commands/uplink/setSerialPort.ts (hunk 2 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/uplink/setSerialPort.ts:9]: fixed in all places

File: src/obis-observer/commands/uplink/setObserverSingleMode.ts (hunk 1 above)
Comment (DeadByElpy) [reply src/obis-observer/commands/uplink/setObserverSingleMode.ts:8]: fixed in all places

File: src/obis-observer/commands/uplink/updateImageWrite.ts [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+};
+
+
+/**
+ * Decode command parameters.
+ *
+ * @returns command payload
+ */
+export const fromBytes = ( bytes: types.TBytes ): ICommandParameters => {
+    if (bytes.length !== COMMAND_BODY_SIZE) {
```
Comment (DarkPark) [inline src/obis-observer/commands/uplink/updateImageWrite.ts:54]:
```suggestion
    if ( bytes.length !== COMMAND_BODY_SIZE ) {
```

Comment (DeadByElpy) [reply src/obis-observer/commands/uplink/updateImageWrite.ts:54]: fixed

**Review by DarkPark — APPROVED** [review]

**Pull request merged**